authN-authZ/auth-istio/README.md

```diff
@@ -45,7 +68,37 @@ kubectl patch deployment -n chatqa <deployment-name> --patch '{
 }'
 ```
 
-The istio ingress gateway will be used to access the chatQnA service in different setups. Follow the istio guide [here](https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/#determining-the-ingress-ip-and-ports) to determine the ingress IP and ports and expose them as environment variables.
+**Ingress Gateway**
+
+The istio ingress gateway will be used to access the chatQnA service in different setups.
+
+First export the router service through istio ingress gateway.
+
+```bash
+kubectl apply -f $(pwd)/$DEPLOY_METHOD/chatQnA_router_gateway.yaml
```

Comment:
This file is only used for option: via JWT token generated by OIDC providers with curl
Why it is defined in the prerequisite section?

Reply:
I think this has no relationship with the token generation way. It is just from istio gateway, and should be added for all megaservices if you want to do authentication.

Reply:
Thanks for your suggestion. Putting the steps in each section will be duplicated. Thus I add "Optional" to this part and add a suggestion for this, do you think it's work?

Reply:
> Thanks for your suggestion. Putting the steps in each section will be duplicated. Thus I add "Optional" to this part and add a suggestion for this, do you think it's work?

Well. I think it is a must for all options to find the ingress port and ip address, but for the step `kubectl apply -f $(pwd)/$DEPLOY_METHOD/chatQnA_router_gateway.yaml`, it is ONLY used for  the first two options of authentication, and NOT needed(instead of optional) for the third option, as it shall only use `kubectl apply -f $(pwd)/$DEPLOY_METHOD/chatQnA_router_gateway_oauth.yaml` instead.

Reply:
Thanks for your explanation!! I have just move this part to "Perform authentication and authorization via JWT tokens generated by OIDC provider" section as your original sequence, please have a check~